# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/localstorage/test/unit/head.js
# Commit: d63dac1ee4d4
# Full Hash: d63dac1ee4d471defff484dae5a3535e1f8ba072
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-10 22:03:45
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Make it possible to initialize temporary origins without ensuring origin directories; r=dom-storage-reviewers,jari
#   
#   LSNG already uses some QuotaManager APIs to achieve that origin directories are
#   not created if they don't exist during datastore preparation, but the feature
#   is not easy to use and it's also not generalized enough for use in other quota
# ==============================================================================

diff -r 2ab08c49f2b1 -r d63dac1ee4d4 dom/localstorage/test/unit/head.js
--- a/dom/localstorage/test/unit/head.js	Thu Oct 10 12:22:54 2024 +0000
+++ b/dom/localstorage/test/unit/head.js	Thu Oct 10 12:22:55 2024 +0000
@@ -102,8 +102,16 @@
   return Services.qms.initializePersistentOrigin(principal);
 }
 
-function initTemporaryOrigin(persistence, principal) {
-  return Services.qms.initializeTemporaryOrigin(persistence, principal);
+function initTemporaryOrigin(
+  persistence,
+  principal,
+  createIfNonExistent = true
+) {
+  return Services.qms.initializeTemporaryOrigin(
+    persistence,
+    principal,
+    createIfNonExistent
+  );
 }
 
 function getOriginUsage(principal) {